core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -804,7 +804,7 @@ static List<DeterministicKeyChain> fromProtobuf(List<Protos.Key> keys, @Nullable
      * key rotation it can happen that there are multiple chains found.
      */
     public static List<DeterministicKeyChain> fromProtobuf(List<Protos.Key> keys, @Nullable KeyCrypter crypter, KeyChainFactory factory) throws UnreadableWalletException {
-        List<DeterministicKeyChain> chains = new LinkedList<>();
```

Comment:
Given that we generally do paste `final` to every variable (even if used as a value), I wonder if this change is in line with the codebase?

Reply:
Given that this particular method has 170 lines of code and has many local variables that are mutated in complex ways, I thought adding `final` here was worth doing. The other 6 variables declared before the for loop are all mutated inside the loop.

Reply:
I agree we don't need `final` here.